library/ziti_ctrl.c

```diff
@@ -64,18 +64,27 @@ struct ctrl_resp {
     void (*ctrl_cb)(void*, ziti_error*, struct ctrl_resp*);
 };
 
+static void ctrl_default_cb (void *s, ziti_error *e, struct ctrl_resp *resp);
+
 static void ctrl_resp_cb(um_http_req_t *req, int code, um_header_list *headers) {
     struct ctrl_resp *resp = req->data;
     resp->status = code;
-    um_http_hdr *h;
-    LIST_FOREACH(h, headers, _next) {
-        if (strcasecmp(h->name, "Content-Length") == 0) {
-            resp->body = malloc(atol(h->value));
-            break;
-        }
-        if (strcasecmp(h->name, "Transfer-Encoding") == 0 && strcmp(h->value, "chunked") == 0) {
-            resp->resp_chunked = true;
-            resp->body = malloc(0);
+    if (code < 0) {
+        NEWP(err, ziti_error);
+        err->code = strdup("CONTROLLER_UNAVAILABLE");
+        err->message = strdup(uv_strerror(code));
+        ctrl_default_cb(NULL, err, resp);
```

Comment:
i don't see ctrl_default_cb free'ing the err nor the err->code and err->message - is that a leak or is it cleaned up somewhere else?

Reply:
it is up to the code that calls controller to keep track or free results and/or errors